app/src/main/java/com/toshi/view/fragment/newconversation/GroupParticipantsFragment.kt

```
@@ -60,12 +60,20 @@ class GroupParticipantsFragment : Fragment() {
     }
 
     private fun initClickListeners() {
-        closeButton.setOnClickListener { this.activity.onBackPressed() }
+        closeButton.setOnClickListener {
+            viewModel.clearParticipants()
+            this.activity.onBackPressed()
+        }
         clearButton.setOnClickListener { search.text = null }
         next.setOnClickListener { handleNextClicked() }
     }
 
-    private fun handleNextClicked() = (this.activity as ConversationSetupActivity).openGroupSetupFlow(viewModel.selectedParticipants.value!!)
+    private fun handleNextClicked() {
+        if (activity !is ConversationSetupActivity) return
```

This should probably throw an `IllegalStateException` instead of just returning since we only want this to work when the activity is `ConversationSetupActivity`.

I think you also can make this cleaner with `when`

```
when(activity) {
    is ConversationSetupActivity -> openGroupSetupFlow(viewModel.selectedParticipants.value ?: listOf())
    else -> throw IllegalStateException("YO you must use an ConversationSetupActivity to make this work")
}
```

What do you think?

---

Your suggestion looks like a good solution to me. I'll rewrite! :+1: